plugins/inputs/t128_metrics/retriever.go

```diff
@@ -0,0 +1,232 @@
+package t128_metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+)
+
+// Retriever implements a way to retrieve metrics
+type Retriever interface {
+	RequestCount() int
+	Describe(index int) string
+	CreateRequest(index int, baseURL string) (*http.Request, error)
+	PopulateResponse(index int, acc telegraf.Accumulator, responseMetrics []ResponseMetric, timestamp time.Time)
+}
+
+// NewIndividualRetriever produces an individual retriever
+func NewIndividualRetriever(useIntegerConversion bool, configuredMetrics []ConfiguredMetric) Retriever {
+	requestMetrics := make([]RequestMetric, 0)
+
+	for _, configMetric := range configuredMetrics {
+		parameters := toRequestParameters(configMetric.Parameters)
+
+		for fieldName, fieldPath := range configMetric.Fields {
+			requestMetrics = append(requestMetrics, RequestMetric{
+				ID:             fieldPath,
+				Parameters:     parameters,
+				OutMeasurement: configMetric.Name,
+				OutField:       fieldName,
+			})
+		}
+	}
+
+	return &individualRetriever{
+		metrics:              requestMetrics,
+		useIntegerConversion: useIntegerConversion,
+	}
+}
+
+// NewBulkRetriever creates a new bulk retriever
+func NewBulkRetriever(useIntegerConversion bool, configuredMetrics []ConfiguredMetric) (Retriever, error) {
+	requestMetrics := make([]BulkRequestMetrics, 0)
+
+	for _, configMetric := range configuredMetrics {
+		bulkRequest := BulkRequestMetrics{
+			IDs:            make([]string, 0, len(configMetric.Fields)),
+			Parameters:     toRequestParameters(configMetric.Parameters),
+			OutMeasurement: configMetric.Name,
+			OutFields:      make(map[string]string, len(configMetric.Fields)),
```

Comment:
Are these the `name` for the metric?

Reply:
If I understand what you mean by "name", then yes.

The configuration looks like:
```
output-field-name = "/stats/random/metric"
output-other-field = "/stats/other/metric"
```
When we make requests for metrics, the responses will have the metric IDs but no info about the output names. So, this is the mapping from the 128T metric ID to the config specified ID (field name).

Reply:
Yes that's the `name` I was referring to - thanks for the explanation.